src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -847,3 +857,22 @@ bool CDSConfig::is_dumping_method_handles() {
 }
 
 #endif // INCLUDE_CDS_JAVA_HEAP
+
+// AOT code generation and its archiving is disabled by default.
+// We enable it only in the final image dump after the metadata and heap are dumped.
+// This affects only JITed code because it may have embedded oops and metadata pointers
+// which AOT code encodes as offsets in final CDS archive regions.
+
+static bool _is_dumping_aot_code_enabled = false;
+
+bool CDSConfig::is_dumping_aot_code_enabled() {
+  return _is_dumping_aot_code_enabled;
```

Comment:
Other functions in CDSConfig don't have the `_enabled()` suffix. E.g., `is_dumping_method_handles()`. It doesn't mean we are doing method handle right now, but rather we have the ability to do so.

Since we rarely ask "am I in the middle of dumping X right now", I think adding `_enabled()` will be redundant.

Reply:
Okay.